controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -86,8 +86,6 @@ func (r *AKODeploymentConfigReconciler) initAVI(
 		}
 
 		if obj.Spec.ControllerVersion != version {
-			// patch the adc if the version doesn't match
-			obj.Spec.ControllerVersion = version
```

Comment:
For all `AKODeploymentConfig` this line is hit during `initAVI` call. Why do you think `obj.Spec.ControllerVersion` not set for some?

Reply:
https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/blob/main/controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go#L71, we check if `r.aviClient` is `nil` or not here, so if `r.aviClient` gets init already, it will skip and not init again and hit this line here.

Reply:
What if we first init aviClient with wrong version, then ADC is updated, but then would aviClient be init again?
We updated all ADC but the AVIClient itself is using the wrong version